vector_helpers.go

```diff
@@ -0,0 +1,82 @@
+package cimgui
+
+// implement vector operations
+
+// Vec2:
+
+// Add returns a sum of v and another.
+func (v ImVec2) Add(another ImVec2) ImVec2 {
+	return ImVec2{
+		X: v.X + another.X,
+		Y: v.Y + another.Y,
+	}
+}
+
+// Sub returns the vector v - another.
+func (v ImVec2) Sub(another ImVec2) ImVec2 {
+	return ImVec2{
+		X: v.X - another.X,
+		Y: v.Y - another.Y,
+	}
+}
+
+// Mul returns the vector v*k.
+func (v ImVec2) Mul(k float32) ImVec2 {
+	return ImVec2{
+		X: v.X * k,
+		Y: v.Y * k,
+	}
+}
+
+// Div returns the vector v/k.
+func (v ImVec2) Div(k float32) ImVec2 {
+	return ImVec2{
+		X: v.X / k,
+		Y: v.Y / k,
+	}
+}
+
+// Eq reports whether p and q are equal.
+func (v ImVec2) Eq(q ImVec2) bool {
+	return v.X == q.X && v.Y == q.Y
+}
+
+// Vec4
+
+// Add returns the rectangle r translated by p.
+func (v ImVec4) Add(p ImVec2) ImVec4 {
+	return ImVec4{
+		X: v.X + p.X,
+		Y: v.Y + p.Y,
+		Z: v.Z + p.X,
+		W: v.W + p.Y,
+	}
+}
+
+// Sub returns the vec4 v translated by -p.
+func (v ImVec4) Sub(p ImVec2) ImVec4 {
+	return v.Add(p.Mul(-1))
+}
+
+// Eq reports whether v and r are equal.
+func (v ImVec4) Eq(r ImVec4) bool {
+	return v.X == r.X && v.Y == r.Y && v.Z == r.Z && v.W == r.W
+}
+
+func (v ImVec4) Canon() ImVec4 {
+	if v.X > v.Z {
+		v.X, v.Z = v.Z, v.X
+	}
+
+	if v.Y > v.W {
+		v.Y, v.W = v.W, v.Y
+	}
+
+	return v
+}
+
+func (v ImVec4) In(s ImVec4) bool {
```

Comment:
This is not needed. ImVec4 is not used in "rect" context in Dear ImGui code. Two ImVec2's are used. ImVec4 represents color 95% of the time.

Reply:
>  ImVec4 represents color 95% of the time.

So is there any situation, when you need to use ImVec4 as a rectangle?

if so, it is necessary as well as Eq method, else - I weill remove these three methods

Reply:
> "So is there any situation, when you need to use ImVec4 as a rectangle?"
No. As I said, for representing rects, two ImVec2's are used. ImVec4 is mostly used for colors.
>
>The only time it is used in Dear ImGui for rectangles is in clip_rect. But it's internal and API uses ImRect for it anyway.

ok, so as far as I understand, I can remove `In`, `Eq` an `Canon` methods?